tests/unit/test_clidocs.py

```diff
@@ -406,6 +407,41 @@ def test_includes_global_args_ref_in_html_options(self):
             "global parameters", rendered
         )
 
+    def test_includes_tagged_union_options(self):
+        help_command = self.create_help_command()
+        shape_model = {
+            'type': 'structure',
+            'union': True,
+            'members': {}
+        }
+        tagged_union = StructureShape('tagged_union', shape_model)
+        arg = CustomArgument(name='tagged_union',
+                             argument_model=tagged_union)
+        help_command.arg_table = {'tagged_union': arg}
+        operation_handler = OperationDocumentEventHandler(help_command)
+        operation_handler.doc_option(arg_name='tagged_union',
+                                     help_command=help_command)
+        rendered = help_command.doc.getvalue().decode('utf-8')
+        self.assertIn('Tagged Union', rendered)
+
+    def test_tagged_union_comes_after_docstring(self):
+        help_command = self.create_help_command()
+        shape_model = {
+            'type': 'structure',
+            'union': True,
+            'members': {}
+        }
+        tagged_union = StructureShape('tagged_union', shape_model)
+        arg = CustomArgument(name='tagged_union',
+                             argument_model=tagged_union,
+                             help_text='FooBar')
+        help_command.arg_table = {'tagged_union': arg}
+        operation_handler = OperationDocumentEventHandler(help_command)
+        operation_handler.doc_option(arg_name='tagged_union',
+                                     help_command=help_command)
+        rendered = help_command.doc.getvalue().decode('utf-8')
+        self.assertRegex(rendered, r'FooBar[\s\S]*Tagged Union')
+
```

Comment:
It would be great if we can add a test as well for tagged unions being documented in the output structure of the command. It looks like the tests are only covering input currently.

Reply:
Agreed. I ended up mocking out a `Shape` object for this test because `OperationDocumentEventHandler.doc_output()` calls a surprising number of methods, `_resolve_shape_ref()` being particularly pesky. Open to suggestions though

Reply:
Ah. We have a class for helping with that. I'd recommend trying to use `botocore.models.DenormalizedStructureBuilder` and just set the union property on the shape it generates. We use it here for example: https://github.com/aws/aws-cli/blob/develop/tests/unit/test_shorthand.py#L166 and helps with all of the scaffolding needed.